languages.adoc

```diff
@@ -20,76 +20,86 @@ SPDX-License-Identifier: Apache-2.0
 
 
 == Requirements
 
 === Content
 
-[.specitem,oft-sid="req~up-language-api~1",oft-needs="impl,utest"]
+A language library
+
+[.specitem,oft-sid="req~up-language-transport-api~1",oft-needs="impl",oft-title="Library declares Transport Layer API"]
 --
-* *MUST* implement serializers, validators, and builders for the link:../basics/README.adoc[uProtocol data types] using the link:../up-core-api/README.adoc[up-core-api data model].
+* *MUST* declare the language specific link:up-l1/README.adoc[uProtocol Transport Layer interface], such that the interface can be implemented by the transport libraries.
 --
 
-[.specitem,oft-sid="req~up-language-transport-api~1",oft-needs="impl,utest"]
+[.specitem,oft-sid="req~up-language-comm-api~1",oft-needs="impl",oft-title="Library declares Communication Layer API"]
 --
-* *MUST* declare the language specific link:up-l1/README.adoc[uProtocol Transport Layer interface], such that the interface can be implemented by the transport libraries.
+* *MUST* declare the language specific link:up-l2/README.adoc[uProtocol Communication Layer (uP-L2) API], such that the interface can be implemented by the transport libraries.
 --
 
-[.specitem,oft-sid="req~up-language-comm-api~1",oft-needs="impl,utest"]
+[.specitem,oft-sid="req~up-language-comm-api-default-impl~1",oft-needs="impl,utest",oft-title="Library implements Communication Layer API"]
 --
-* *MUST* declare the language specific link:up-l2/README.adoc[uProtocol Communication Layer (uP-L2) API] and provide a default implementation based on the (abstract) Transport Layer API.
+* *MUST* provide a default implementation of link:up-l2/README.adoc[uProtocol Communication Layer (uP-L2) API] based on the (abstract) Transport Layer API.
 --
 
-[.specitem,oft-sid="req~up-language-naming~1",oft-needs="impl,utest"]
+[.specitem,oft-sid="dsn~up-language-naming~1",oft-title="Library follows naming pattern"]
 --
-* *MUST* be named according to pattern `up-[LANGUAGE]`
+* *MUST* be named according to pattern `up-[LANGUAGE]`, e.g. `up-rust` or `up-python`.
 --
 
-[.specitem,oft-sid="req~up-language-structure~1",oft-needs="impl,utest"]
+[.specitem,oft-sid="dsn~up-language-structure~1",oft-title="Library uses uProtocol namespace"]
 --
-* The top level namespace (package name) *MUST* contain `uprotocol` and *SHOULD* contain `org.eclipse.uprotocol` 
+* *MUST* use the programming language specific namespace concept, if availabe, to indicate that the library belongs to the _uProtocol_ project. If the language supports hierarchical namespaces (like Java does), then `org.eclipse.uprotocol` *SHOULD* be used. Otherwise, a namespace of `uprotocol` *MUST* be used*.
 --
 
-[.specitem,oft-sid="req~up-language-documentation~1",oft-needs="impl,utest"]
+[.specitem,oft-sid="req~up-language-documentation~1",oft-needs="uman",oft-title="Library contains adequate README file"]
```

Comment:
uman is a new one but I gather it refers to user manual

Reply:
it does